### PR TITLE
Make building BBL image dependent on UseRiscVBBL

### PR DESCRIPTION
**Pull request by axel-h**
Signed-off-by: Axel Heider <axelheider@gmx.de>

### CODE REVIEW

File: cmake-tool/helpers/rootserver.cmake [hunk 1]
```diff
@@ -30,7 +30,7 @@ mark_as_advanced(TLS_ROOTSERVER)
 find_file(UIMAGE_TOOL make-uimage PATHS "${CMAKE_CURRENT_LIST_DIR}" CMAKE_FIND_ROOT_PATH_BOTH)
 mark_as_advanced(UIMAGE_TOOL)
 
-if(KernelArchRiscV)
+if(KernelArchRiscV AND KernelRiscVBBL)
```
Comment (yyshen) [inline cmake-tool/helpers/rootserver.cmake:33]: If the KernelRiscVBBL depends on KernelArchRiscV, the first condition is probably not needed.

File: cmake-tool/helpers/rootserver.cmake [hunk 2]
```diff
@@ -90,48 +90,43 @@ function(DeclareRootserver rootservername)
...
-                    --host=${host}
-                    --with-arch=${march}
-                    --with-payload=${elf_target_file}
-                        && make -s clean && make -s > /dev/null
-                DEPENDS ${elf_target_file} elfloader ${USES_TERMINAL_DEBUG}
-            )
-            set(elf_target_file "${CMAKE_BINARY_DIR}/bbl/bbl")
+            if(KernelRiscVBBL)
+                # Package up our final elf image into the Berkeley boot loader.
+                # The host string is extracted from the  cross compiler setting
```
Comment (yyshen) [inline cmake-tool/helpers/rootserver.cmake:100]: two spaces between "the" and "cross"

Comment (axel-h) [reply cmake-tool/helpers/rootserver.cmake]: > two spaces between "the" and "cross"

fixed

File: cmake-tool/helpers/rootserver.cmake (hunk 1 above)
Comment (axel-h) [reply cmake-tool/helpers/rootserver.cmake]: > If the KernelRiscVBBL depends on KernelArchRiscV, the first condition is probably not needed.

fixed

**Comment by axel-h** [comment]
I've withdrawn https://github.com/seL4/seL4/pull/194 and added a `UseRiscVBBL` here now. Things works fine for me now when I put a `set(UseRiscVBBL OFF)` in my `kernel/src/plat/${my-risc-v-chip}/config.cmake`, the build will not try to package the ELFLoder into BBL then.

**Comment by yyshen** [comment]
> I've withdrawn [seL4/seL4#194](https://github.com/seL4/seL4/pull/194) and added a `UseRiscVBBL` here now. Things works fine for me now when I put a `set(UseRiscVBBL OFF)` in my `kernel/src/plat/${my-risc-v-chip}/config.cmake`, the build will not try to package the ELFLoder into BBL then.
> 
> Any suggestions on a the name "UseRiscVBBL" are welcome.

You added the option here instead of the kernel's CMake file. I suggest RiscVMModeBBL, RiscVMachModeBBL or RiscVMachineModeBBL. RiscVMachineModeBBL is a bit long, but I think it expresses the purpose clearly. What do you think?

**Comment by axel-h** [comment]
What about` RiscVMModeBBL`? Or we simply keep `UseRiscVBBL` for now. Does seL4 have something similar for ARM's ATF integration we could adopt?

**Comment by yyshen** [comment]
> What about` RiscVMModeBBL`? Or we simply keep `UseRiscVBBL` for now. Does seL4 have something similar for ARM's ATF integration we could adopt?

RiscVMModeBBL and UseRiscVBBL are both fine to me. I do not think we have something similar for Arm.

**Comment by yyshen** [comment]
@ssrg-bamboo 'test'

**Comment by yyshen** [comment]
@LukeMondy Should I still use ssrg-bamboo `test` for a build system change test?

**Comment by lsf37** [comment]
I think so, yes. Something doesn't seem to be working for summoning the bot. Let me try:
@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
All the tests we ran have passed! Nice job!

**Comment by yyshen** [comment]
> I think so, yes. Something doesn't seem to be working for summoning the bot. Let me try:
> 
> @ssrg-bamboo `test`

well, you are much more powerful, the bot is up.

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by lsf37** [comment]
> well, you are much more powerful, the bot is up.

yeah, that worries me a bit, the bot should listen to you exactly the same. @LukeMondy, can you have a look at this on Mon, please?

**Comment by LukeMondy** [comment]
@yyshen You're initial comment was 
> @ssrg-bamboo 'test'

which was then edited to use the `code backticks` around test. The bot works on notifications, and it seems GitHub doesn't notify (at least in the same way) when an edit occurs on a comment.

You can see from your later comment, where you quoted Gerwin, that your comment did invoke the bot, so you do have the proper powers :pray: 

I'll soon have some more time to work on improving the bot, particularly so it will explain when things go wrong.

**Comment by yyshen** [comment]
> @yyshen You're initial comment was
> 
> > @ssrg-bamboo 'test'
> 
> which was then edited to use the `code backticks` around test. The bot works on notifications, and it seems GitHub doesn't notify (at least in the same way) when an edit occurs on a comment.
> 
> You can see from your later comment, where you quoted Gerwin, that your comment did invoke the bot, so you do have the proper powers 
> 
> I'll soon have some more time to work on improving the bot, particularly so it will explain when things go wrong.

OK, I will try to remember using backticks.

**Comment by lsf37** [comment]
Thanks, @LukeMondy, well spotted!

**Pull request merged**